Player.java

```diff
@@ -17,23 +17,27 @@ public Player( String name, int dmg, int hp, double as, int xppl, int dmgipl, in
     }
 
     public void attack(Unit enemy) {
-        if (enemy.isAlive() && this.isAlive()) {
-            enemy.setHP(enemy.getHp()-this.getDMG());
+        if(enemy.hp<this.dmg){
+            this.xp += enemy.hp;
+            super.attack(enemy);
+            this.lvlup();
+        } else {
+            super.attack(enemy);
+            this.xp +=this.getDMG();
+            this.lvlup();
```

Comment:
rendben van, de a dmg < > hp logika most ugyanugy megjelenik itt, mint a Unit.attack-ban. Egy szebb megoldas lenne, ha az a fv visszaterne a bevitt sebzes ertekevel, itt meg csak annyi lenne, hogy `this.xp+=super.attack(enemy)` Es akkor az, hogy meg kell kulonboztetni ket esetet, az csak egy helyen van lekezelve.